Samples/ControlMonitorApp/CameraControlMonitorAppSample/MainPage.xaml.cs

```diff
@@ -0,0 +1,206 @@
+                UICameraList.Visibility = Visibility.Visible;
+            });
+            base.OnNavigatedTo(e);
+        }
+
+        protected override void OnNavigatedFrom(NavigationEventArgs e)
+        {
+            UninitMediaCapture();
+            base.OnNavigatedFrom(e);
+        }
+
+        private async Task InitializeAsync(int selectedIndex)
+        {
+            m_mediaCapture = new MediaCapture();
+            m_mediaPlayer = new MediaPlayer();
+
+            // We initialize the MediaCapture instance with the camera in sharing mode
+            // to preview its stream without blocking other app from using it
+            var initSettings = new MediaCaptureInitializationSettings()
+            {
+                // This app could define "SharingMode= MediaCaptureSharingMode.SharedReadOnly" here if
+                // the app only wants to be informed of the changes and not have full control camera app.
+                VideoDeviceId = m_cameraDeviceList[selectedIndex].Id,
+                StreamingCaptureMode = StreamingCaptureMode.Video
+            };
+
+            await m_mediaCapture.InitializeAsync(initSettings);
+
+            // Retrieve the source associated with the video preview stream.
+            // On 1-pin camera, this may be the VideoRecord MediaStreamType as opposed to VideoPreview on multi-pin camera
+            var frameSource = m_mediaCapture.FrameSources.FirstOrDefault(source => source.Value.Info.MediaStreamType == MediaStreamType.VideoPreview
+                                                                              && source.Value.Info.SourceKind == MediaFrameSourceKind.Color).Value;
+            if (frameSource == null)
+            {
+                frameSource = m_mediaCapture.FrameSources.FirstOrDefault(source => source.Value.Info.MediaStreamType == MediaStreamType.VideoRecord
+                                                                                  && source.Value.Info.SourceKind == MediaFrameSourceKind.Color).Value;
+            }
+
+            // if no preview stream is available, bail
+            if (frameSource == null)
+            {
+                return;
+            }
+
+            // Setup MediaPlayer with the preview source
+            m_mediaPlayer.RealTimePlayback = true;
+            m_mediaPlayer.AutoPlay = true;
+            m_mediaPlayer.Source = MediaSource.CreateFromMediaFrameSource(frameSource);
+
+            UIMediaPlayerElement.SetMediaPlayer(m_mediaPlayer);
+
+            // Creating a controls data for listening changes on these controls.
+            var controlContrast = new ControlMonitorHelper.ControlData()
+            {
+                controlKind = ControlMonitorHelper.ControlKind.VidCapVideoProcAmpKind,
+                controlId = (UInt32)CameraKsPropertyHelper.VidCapVideoProcAmpKind.KSPROPERTY_VIDEOPROCAMP_CONTRAST
+            };
+
+            var controlBrightness = new ControlMonitorHelper.ControlData()

```

Comment:
It looks like we are always writing the Brightness control without considering the EV Compensation control.

So, on a camera that supports EV Compensation and Brightness, the app would be changing a different control than the Settings app is changing, and the sync won't work.

Can we update the logic here to match what the Settings app does? Basically it is:

```
if (Camera supports EV Compensation)
   Brightness slider controls EV Compensation
elif (Camrea supports Brightness)
   Brightness slider controls Brightness
else
   Brightness slider not shown
```

Reply:
added